### DISCUSSION TITLE
[BUG] Photos listed under the incorrect date section

### DISCUSSION THREAD

**Discussion by ghost** [immich-app/immich#12323] [Q&A]
### The bug

the iOS shows the photos correctly for 8th Nov, 1st Nov and 31 Aug

but the web shows 8th Nov and all others in the 31 Aug (including the 1st Nov photo)

# web
<img width="1664" alt="Screenshot 2023-11-08 at 10 32 02 pm" src="https://github.com/immich-app/immich/assets/1471841/53e4cbbf-4dd9-42c8-9fc8-82b7b5e60719">

# web photo info showing date
<img width="312" alt="Screenshot 2023-11-08 at 10 41 37 pm" src="https://github.com/immich-app/immich/assets/1471841/c80a8b9e-2aec-448c-b191-9ad7dd6cf6a5">

# iOS view correct
![IMG_3450](https://github.com/immich-app/immich/assets/1471841/e459a5ec-2367-418d-9f1e-ea1ada7be22c)

### The OS that Immich Server is running on

ubuntu 22

### Version of Immich Server

1.85

### Version of Immich Mobile App

1.85

### Platform with the issue

- [ ] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
default
```

### Your .env content

```Shell
default
```

### Reproduction steps

```bash
1.
2.
3.
...
```

**Comment by atlas-shrugged08** [comment]
fwiw, I am (on v1.84) seeing the same issues. (just do not have the bandwidth to anonymize screenshot). 

I additionally see the sequence of dates wrong on web. e.g. Oct 22 shows before Oct 25. (for which i can show a non anonymized screenshot below)
![image](https://github.com/immich-app/immich/assets/123214845/d385ccfd-fb47-424c-af3c-0835ab94bd33)

**Comment by andyxpert** [comment]
There is somewhat a problematic area when dealing with dates... 
Normally a date can be taken from multiple areas:
- file properties (Created, Modified, Access dates) - these are auto-updated by the OS whenever a file action is performed. Keep in mind that changing the file, renaming it, moving to a different folder will shortcut Immich's DB and normally it should be refreshed once the job runs (Library or Extract Metadata)
- EXIF data - here it's a mess as there are many exif structures/properties and based on what device creates the original photo some of the fields may be missing or may contain wrong data...
- Cache or DB stored Metadata - it may happen that Immich uses some cached data or pre-scanned data that may not reflect the reality

I have a big photo collection (around 100k files) taken throughout 10+ years with different devices (phones, cameras, DSLR, downloads from FB/WA, etc.), some processed by other 3rd party software (each causing changes to the EXIF data), moved from folder to folder, HDD backups, etc.... Long-story-short, the dates may not be that accurate for all of them, obviously :)

I do not blame Immich for these issues, though a view of filtered photos (based on date ranges or location) where we can batch-update dates would be God-sent :) (maybe I'll create a feature-request with this...)

What I suggest (my approach that works 100% and it's fully controlled by me) is you create some bash scripts that correct the dates for the chunks of your library that have problems...
Something like this:
- iterate through the chunk of photos where you want to change the dates
- grab the exif dates and grab the file properties dates (sometimes the creation date is the source-of-truth, sometimes the exif original date/time is the source of truth)
- evaluate the dates (see which one is the correct one)
- alter the file by setting that date (with "touch" to set the modified date, or with exiftool to change the exif dates)
- ask Immich to rescan / reimport that chunk (based on location)

I'm currently in the process of cleaning by library - I'm already 1 week into this process... last 3 days it kept on generating thumbnails... but I don't see any wrongly placed photos in the timeline, so I guess my logic is good :)

TL;DR: try to isolate the files that are misplaced (as the dates aren't accurate) and deal with them manually before re-importing them into Immich and this should fix your problem.

Immich guys, anything you can do to help in this situation ? I would LOVE to ditch my scripts and have Immich do it by default :)

**Comment by ghost** [comment]
@andyxpert  shown in my issue post, it clearly shows the info pane of the photo within immich itself, with the date 1st Nov.. but it is listing it under August in web viewer

This seems to be a immich ordering problem.. it orders correctly in iOS but not correctly in the web, again as shown in the images in the post

batch scripts is not a solution for this..

**Comment by andyxpert** [comment]
Well, each with their own scenarios, what works for one may not be applicable to others...
If iOS app shows something and web app shows something else, then indeed it's a Bug.

**Comment by Tyree** [comment]
Mine is the opposite. Images shown on the correct date (Nov 18) on the web app, but on the Android app, photos are randomly spread out from Nov 4-18

**Comment by AngelPone** [comment]
In my case, some photos shown on some future dates on iOS app, but show in Chrome correctly. 
One interesting point is that I use the "change date" functionality on the Web app. I changed the date to 2017-02-25, but they shown on 2024-02-25, so weird.

**Comment by jrasm91** [comment]
If you have a sample photo with this issue, please upload here as a zip or send it to me via email or discord and I will take a look.
